EIPS/eip-6229.md

```diff
@@ -0,0 +1,335 @@
+    - name: redeemAssets
+    - type: uint256
+```
+
+#### getScheduledDeposits
+
+Get the `totalAssets` of scheduled deposits for `depositor`.
+
+MUST *NOT* revert.
+
+```
+- name: getScheduledDeposits
+  type: function
+  stateMutability: view
+
+  inputs:
+    - name: depositor
+    - type: address
+
+  outputs:
+    - name: totalAssets
+    - type: uint256
+```
+
+#### getScheduledRedemptions
+
+Get the `totalShares` of scheduled redemptions for `redeemer`.
+
+MUST *NOT* revert.
+
+```
+- name: getScheduledRedemptions
+  type: function
+  stateMutability: view
+
+  inputs:
+    - name: redeemer
+    - type: address
+
+  outputs:
+    - name: totalShares
+    - type: uint256
+```
+
+### Events
+
+#### ScheduleDeposit
+
+`sender` schedules a deposit with `assets` in this `round`.
+
+MUST be emitted via `scheduleDeposit` method.
+
+```
+- name: ScheduleDeposit
+  type: event
+
+  inputs:
+    - name: sender
+      indexed: true
+      type: address
+    - name: assets
+      indexed: false
+      type: uint256
+    - name: round
+      indexed: false
+      type: uint256
+```
+
+#### ScheduleRedeem
+
+`sender` schedules a redemption with `shares` in this `round`.
+
+MUST be emitted via `scheduleRedeem` method.
+
+```
+- name: ScheduleRedeem
+  type: event
+
+  inputs:
+    - name: sender
+      indexed: true
+      type: address
+    - name: shares
+      indexed: false
+      type: uint256
+    - name: round
+      indexed: false
+      type: uint2
+```
+
+#### SettleDeposits
+
+Settle scheduled deposits for `depositor` in this `round`. Issue `newShares` and transfer them to the `depositor`.
+
+MUST be emitted via `settleDeposits` method.
+
+```jsx
+- name: SettleDeposits
+  type: event
+
+  inputs:
+    - name: depositor
+      indexed: true
+      type: address
+    - name: newShares
+      type: uint256
+    - name: round
+      type: uint256
+```
+
+#### SettleRedemptions
+
+Settle scheduled redemptions for `redeemer` in this `round`. Burn `burnShares` and transfer `redeemAssets` back to the `redeemer`.
+
+MUST be emitted via `settleRedemptions` method.
+
+```
+- name: SettleRedemptions
+  type: event
+
+  inputs:
+    - name: redeemer
+      indexed: true
+      type: address
+    - name: burnShares
+      type: uint256
+    - name: redeemAssets
+      type: uint256
+    - name: round
+      type: uint256
+```
+
+## Rationale
+
+The standard is designed to be a minimal interface. Details such as the start and end of a lock-in period, and how the underlying tokens are being used during the lock-in period are not specified.
+
+There is no function for scheduling a withdrawal, since during the lock-in period, the share price is undetermined, so it’s not able to decide how many underlying tokens can be withdrawn.
+
+## Backwards Compatibility
+
+The `deposit`, `mint`, `withdraw`, `redeem` methods for [EIP-4626](./eip-4626.md) should be reverted when vault is in `LOCKED` state to prevent issuing or burning shares with an undefined share price.
```

Comment:
```suggestion
The `deposit`, `mint`, `withdraw`, `redeem` methods for [EIP-4626](./eip-4626.md) should revert when a vault is in the `LOCKED` state to prevent issuing or burning shares with an undefined share price.
```